src/main/java/io/github/genomicdatainfrastructure/daam/utils/ApplicationValidator.java

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.daam.utils;
+
+import io.github.genomicdatainfrastructure.daam.exceptions.ApplicationNotFoundException;
+import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationsApi;
+import io.github.genomicdatainfrastructure.daam.remote.rems.model.Application;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.WebApplicationException;
+
+@ApplicationScoped
+public class ApplicationValidator {
+
+    public Application validate(Long applicationId, String remsApiKey, String userId,
```

Comment:
important: this class is doing much more than validating, it's an strategy/adapter to fetch Applications from REMS. Perhaps this one should have a generic interface, and the mapper should be part of it, as composition.

Let's validate this idea together.